packages/flutter/lib/src/material/flexible_space_bar.dart

```diff
@@ -468,7 +468,7 @@ class _RenderFlexibleSpaceHeaderOpacity extends RenderOpacity {
     if (child == null) {
       return;
     }
-    if (opacity == 0) {
+    if (opacity == 0 || !needsCompositing) {
```

Comment:
So the problem here is that when we round opacity to an integer, we're likely cutting off small values and so the compositing states are disagreeing. However, we technically don't need to composite at fully opaque (though we do today). In the future, changes to this compositing behavior may make this assumption that !needsCompositing = drop the layer incorrect.

I think it would be safer to either completely reuse the RenderOpacity paint method (if possible), or repeat the same rounding procedure for opacity here.

Reply:
If we make line above instead `(opacity * 255).roundToDouble() <= 0`, does this problem occur?

Reply:
My thought was to round the the value as well but I was thinking it might round this value too many times while scrolling and then i checked what `needsCompositing` status is and it turned it to be false so I used it insterad.

Thanks for the insightful comment. This particular area is new for me so I learned some stuff from this small PR.